tyr/servers/cache/server.py

```diff
@@ -0,0 +1,152 @@
+
+        # This is just a temporary fix to override the default security
+        # groups for Cache servers until the security_groups argument is
+        # removed.
+
+        self.security_groups = [
+            'management',
+            'chef-nodes',
+            self.envcl,
+            '{env}-cache-management'.format(env=self.environment[0])
+        ]
+
+        self.resolve_security_groups()
+
+    def bake(self):
+
+        super(CacheServer, self).bake()
+
+        if not any([self.couchbase_version, self.couchbase_username,
+                        self.couchbase_password]):
+            return
+
+        with self.chef_api:
+
+            if self.couchbase_version:
+                self.chef_node.attributes.set_dotted('couchbase.server.version',
+                                                        self.couchbase_version)
+
+                self.log.info('Set the couchbase.server.version to' \
+                            '{version}'.format(version=self.couchbase_version))
+
+            if self.couchbase_username:
+                self.chef_node.attributes.set_dotted('couchbase.server.username',
+                                                        self.couchbase_username)
+
+                self.log.info('Set the couchbase.server.username to' \
+                            '{username}'.format(username=self.couchbase_username))
+
+            if self.couchbase_password:
+                self.chef_node.attributes.set_dotted('couchbase.server.password',
+                                                        self.couchbase_password)
+
+                self.log.info('Set the couchbase.server.password to' \
+                            '{password}'.format(password=self.couchbase_password))
+
+            self.chef_node.save()
+            self.log.info('Saved the Chef Node configuration')
+
+    def configure_couchbase(self):
+
+        with self.chef_api:
+
+            self.chef_node = chef.Node(self.name)
+
+            username = self.chef_node.attributes.get_dotted(
+                                                'couchbase.server.username')
+            password = self.chef_node.attributes.get_dotted(
+                                                'couchbase.server.password')
+
+            memory_quota = self.chef_node.attributes.get_dotted(
+                                             'couchbase.server.memory_quota_mb')
+
+            port = self.chef_node.attributes.get_dotted('couchbase.server.port')
+
+            uri = 'http://{hostname}:{port}/pools/default/buckets'.format(
+                                                hostname = self.hostname,
+                                                port = port)
+
+            payload = {
+                'authType': 'sasl',
+                'bucketType': 'memcached',
+                'name': self.bucket_name,
+                'ramQuotaMB': memory_quota
+            }
+
+            auth = (username, password)
+
+            while True:
```

Comment:
Yeah, it would loop until it managed to successfully connect to the Couchbase server. It's something we already do, specifically when managing the SSH connection and for automating stuff for MongoDB instances. I think it's fine since we're monitoring it and it's logging what's happening so it doesn't look like it's just stuck. The user would be aware of what's happening.